tests/parametric/test_sampling_span_tags.py

```diff
@@ -0,0 +1,118 @@
+                    name="child", service="webserver", parent_id=parent_span.span_id
+                ) as child_span:
+                    child_span.set_meta(MANUAL_KEEP_KEY, None)
+
+        traces = test_agent.wait_for_num_spans(2, clear=True)
+
+        parent_span = find_span_in_traces(traces, Span(name="parent", service="webserver"))
+        child_span = find_span_in_traces(traces, Span(name="child", service="webserver"))
+
+        _assert_sampling_tags(parent_span, child_span, None, "-3", 2, 1)
+
+    @bug(library="python", reason="Python sets dm tag on child span")
+    @bug(library="python_http", reason="Python sets dm tag on child span")
+    @bug(library="php", reason="PHP sets dm tag -1 on parent span")
+    @bug(library="golang", reason="golang sets dm tag -1 on parent span")
+    @bug(library="java", reason="java sets dm tag -1 on parent span")
+    def test_tags_defaults_sst002(self, test_agent, test_library):
+        parent_span, child_span = _get_parent_and_child_span(test_agent, test_library)
+        _assert_sampling_tags(parent_span, child_span, None, "-0", 1, None)
+
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_SAMPLE_RATE": 1}])
+    @bug(library="python", reason="Python sets dm tag on child span")
+    @bug(library="python_http", reason="Python sets dm tag on child span")
+    def test_tags_defaults_rate_1_sst003(self, test_agent, test_library):
+        parent_span, child_span = _get_parent_and_child_span(test_agent, test_library)
+        _assert_sampling_tags(parent_span, child_span, None, "-3", 2, 1)
+
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_SAMPLE_RATE": 1e-06}])
+    @bug(library="php", reason="PHP sets dm tag -1 on parent span")
+    @bug(library="java", reason="Java sets rate tag 9.9999 on parent span")
+    @bug(library="dotnet", reason="Dotnet sets rate tag 9.9999 on parent span")
+    def test_tags_defaults_rate_tiny_sst004(self, test_agent, test_library):
+        parent_span, child_span = _get_parent_and_child_span(test_agent, test_library)
+        _assert_sampling_tags(parent_span, child_span, None, None, -1, 1e-06)
+
+    @pytest.mark.parametrize(
+        "library_env", [{"DD_TRACE_SAMPLE_RATE": 1, "DD_TRACE_SAMPLING_RULES": json.dumps([{"sample_rate": 1}])}]
+    )
+    @bug(library="python", reason="Python sets dm tag on child span")
+    @bug(library="python_http", reason="Python sets dm tag on child span")
+    def test_tags_defaults_rate_1_and_rule_1_sst005(self, test_agent, test_library):
+        parent_span, child_span = _get_parent_and_child_span(test_agent, test_library)
+        _assert_sampling_tags(parent_span, child_span, None, "-3", 2, 1)
+
+    @pytest.mark.parametrize(
+        "library_env", [{"DD_TRACE_SAMPLE_RATE": 1, "DD_TRACE_SAMPLING_RULES": json.dumps([{"sample_rate": 0}])}]
+    )
+    @bug(library="golang", reason="golang sets dm tag on parent span")
+    def test_tags_defaults_rate_1_and_rule_0_sst006(self, test_agent, test_library):
```

Comment:
Sampling tests are difficult to follow and maintain. Adding a doc string to each test case could be helpful.